packages/data-source/src/DataSource.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SubmissionHandler } from '@elyra/application';
+
+export interface ICodeTemplate {
+  language: string;
+  framework: string;
+  code: string[];
+}
+
+export interface IDataSource {
+  name: string;
+  id: number;
+  displayName: string;
+  description: string;
+  source: string;
+  language: string;
+  code: ICodeTemplate[];
+}
+
+export class DataSourceManager {
+  readonly dataSourceEndpoint = 'api/metadata/data-sources';
+
+  async findAll(): Promise<IDataSource[]> {
+    const getDataSources: Promise<IDataSource[]> = new Promise(
+      (resolve, reject) => {
+        const allDataSources: IDataSource[] = [];
+        let idNum = 0;
+        SubmissionHandler.makeGetRequest(
+          this.dataSourceEndpoint,
+          'data sources',
+          (response: any) => {
+            const dataSourcesResponse = response['data-sources'];
+
+            for (const dataSourceKey in dataSourcesResponse) {
+              const jsonDataSource = dataSourcesResponse[dataSourceKey];
+              idNum++;
+              const dataSource: IDataSource = {
+                name: jsonDataSource.name,
+                id: idNum,
+                displayName: jsonDataSource.display_name,
+                description: jsonDataSource.metadata.description,
+                source: jsonDataSource.metadata.source,
+                language: jsonDataSource.metadata.language,
```

Comment:
It seems like we'll be using these based on language.  Should we go ahead and store these relative to language now so that we can just index by language later?

If so, I would normalize the language (lower case) as the index, but not necessarily normalize _this_ value - which is likely used for display purposes.  That way, languages like `R` will still appear in their native case (or however the user created the data source in the first place).

`FindByLanguage` would then just normalize its parameter and return the data sources relative to that language (or do whatever if language isn't in the list).